src/win32/waiter.cpp

```diff
@@ -125,8 +125,7 @@ int WinWaiter::wait()
         r |= flags[dwWaitResult - WAIT_OBJECT_0];
     }
 
-    handles.clear();
-    flags.clear();
+    index = 0;
```

Comment:
I don't know if there could be a huge amount of handles in `handles` temporary, but in that case maybe we want to shrink the vector from time to time. Imagine that only temporary we wait for 20.000 objects, but the average is in the hundreds. We would keep keep memory for the maximum amount of handles ever tracked, which is not that much, anyway (20k*sizeof(HANDLE)=156KB is probably acceptable, anyway)
Apart from that memory comsumption concern, the PR looks totally fine :)

Reply:
Yes, I thought about that too, but thought keeping the PR simple and just concentrating on performance for now is the way to go, memory for this one won't be too bad and we can adjust it later if it turns out to be an issue.  thanks!